code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/EnsureSettingForProvider.java

```diff
@@ -0,0 +1,80 @@
+package io.cattle.platform.iaas.api.auth.identity;
+
+import io.cattle.platform.api.auth.Identity;
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.iaas.api.auth.SecurityConstants;
+import io.cattle.platform.iaas.api.auth.SettingsUtils;
+import io.cattle.platform.iaas.api.auth.integration.github.GithubConstants;
+import io.cattle.platform.iaas.api.auth.integration.github.GithubUtils;
+import io.cattle.platform.util.type.InitializationTask;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class EnsureSettingForProvider implements InitializationTask {
+
+    private static final Log logger = LogFactory.getLog(EnsureSettingForProvider.class);
+
+    @Inject
+    SettingsUtils settingsUtils;
+
+    @Inject
+    GithubUtils githubUtils;
+
+    @Override
+    public void start(){
+        if (SecurityConstants.SECURITY.get() &&
+                StringUtils.isNotBlank(GithubConstants.GITHUB_CLIENT_SECRET.get()) &&
+                StringUtils.isNotBlank(GithubConstants.GITHUB_CLIENT_ID.get())){
+            if (StringUtils.isBlank(SecurityConstants.AUTH_PROVIDER.get())) {
+                settingsUtils.changeSetting(SecurityConstants.AUTH_PROVIDER_SETTING, GithubConstants.CONFIG);
+                logger.info("Upgrade detected. Adding setting for :" + SecurityConstants
+                        .AUTH_PROVIDER_SETTING);
+            }
+            String userSetting = "api.auth.github.allowed.users";
+            String orgSetting = "api.auth.github.allowed.orgs";
+            String allowedUsers = ArchaiusUtil.getString(userSetting).get();
+            String allowedOrgs = ArchaiusUtil.getString(orgSetting).get();
+            if ("restricted".equalsIgnoreCase(GithubConstants.ACCESS_MODE.get()) &&
+                    StringUtils.isBlank(GithubConstants.GITHUB_ALLOWED_IDENTITIES.get()) &&
+                    (StringUtils.isNotBlank(allowedOrgs) || StringUtils.isNotBlank(allowedUsers))
+                    ){
+                List<Identity> allowedIdentities = new ArrayList<>();
+                List<String> userIds = githubUtils.fromCommaSeparatedString(allowedUsers);
+                for (String userId: userIds){
+                    Identity identity = new Identity(GithubConstants.USER_SCOPE, userId.split(":")[1]);
```

Comment:
Minor, but it's possible to hit an indexOutOfRange exception if userId doesn't have a ':' character.